src/cargo/core/summary.rs

```diff
@@ -431,33 +429,63 @@ impl fmt::Display for FeatureValue {
 
 pub type FeatureMap = BTreeMap<InternedString, Vec<FeatureValue>>;
 
-fn validate_feature_name(config: &Config, pkg_id: PackageId, name: &str) -> CargoResult<()> {
+fn validate_feature_name(pkg_id: PackageId, name: &str) -> CargoResult<()> {
```

Comment:
It looks like mostly a duplicate of this (slightly different though)

https://github.com/rust-lang/cargo/blob/b81b251d2c87594c836cd14bbe06f8ff3a3a6164/src/cargo/util/restricted_names.rs#L39-L84


Should we merge them or keep it as-is?